File: linkedlist/swapNodesInPairs.java
// https://leetcode.com/problems/swap-nodes-in-pairs/

package linkedlist;

public class swapNodesInPairs{
    public static void main(String[] args) {
        System.out.println("This is an interactive problem - Run Solution in Leetcode ");
    }
}

//Definition for singly-linked list.

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy = new ListNode(-1);
        ListNode prev = dummy, current = head;
        dummy.next = head;

        while (current!=null && current.next !=null){
            prev.next = current.next;
            current.next= current.next.next;
            prev.next.next = current;

            current = current.next;
            prev = prev.next.next;
        }

        return dummy.next;
    }
}

File: linkedlist/swapNodesInPairs.py
# Definition for singly-linked list.

class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
        
def swapPairs(head):
    """
    :type head: ListNode
    :rtype: ListNode
    """
    dummy = ListNode(-1)
    prev, current = dummy, head;
    dummy.next = head
    
    while current and current.next: 
        prev.next = current.next
        current.next= current.next.next
        prev.next.next = current
        
        current = current.next
        prev = prev.next.next
    
    return dummy.next    
    
def printListNode(head):
    while head:
        print(str(head.val) +" ----->", end=" ")
        head= head.next
    print()
    
list = ListNode(1)
list.next = ListNode(2)
list.next.next = ListNode(3)
list.next.next.next = ListNode(4)

printListNode(list)
list = swapPairs(list)
printListNode(list)